disnake/ext/commands/interaction_bot_base.py

```diff
@@ -336,19 +450,72 @@ def add_message_command(self, message_command: InvokableMessageCommand) -> None:
         TypeError
             The message command passed is not an instance of :class:`InvokableMessageCommand`.
         """
-        if not isinstance(self, disnake.Client):
-            raise NotImplementedError("This method is only usable in disnake.Client subclasses")
+        warn_deprecated(
+            "add_message_command is deprecated and will be removed in a future version. "
+            "Use add_app_command instead.",
+            stacklevel=3,
+        )
+        self.add_app_command(message_command)
 
-        if not isinstance(message_command, InvokableMessageCommand):
-            raise TypeError(
-                "The message_command passed must be an instance of InvokableMessageCommand"
-            )
+    def remove_app_command(
+        self, cmd_type: ApplicationCommandType, name: str, guild_ids: Optional[Tuple[int, ...]]
```

Comment:
I couldn't choose between multiple guild IDs and one guild ID, so I decided to choose multiple because this simplifies the code in `Cog._eject`. I agree with the point about ambiguities though

Reply:
Maybe taking a single guild ID makes more sense but it makes it unclear how global commands should be treated by this function when `test_guilds` are specified

Reply:
Hm, good point, not sure either. Global commands make this a bit more difficult, I'll do some tinkering tomorrow to see if maybe there's a nice solution

Reply:
Did some thinking; since all methods automatically translate "global registration" into "test_guilds registration" when `test_guilds` is set, doing the same thing here yields the same set of guilds from which we want to un-register the command.
This should mean that `for id in ids: self.all_app_commands.pop(...)` here returns the same exact `InvokableApplicationCommand` object for all guilds, so we don't lose any information by only returning the first one like implemented right now.

With that in mind, taking a single guild ID parameter shouldn't be ambiguous even in the test_guilds case, I think

Reply:
Fair enough. So we just pop all test guild commands and return the first match then, right? This should work, yeah

Reply:
https://github.com/DisnakeDev/disnake/pull/1107/commits/93880cffb149490166e1c3d9c3d98b77d59a2107